lib/journaled/transaction_ext.rb

```diff
@@ -0,0 +1,82 @@
+require 'active_record/connection_adapters/abstract/transaction'
+
+module Journaled
+  module TransactionExt
+    def before_commit_records
+      @_journaled_committing_records = true
+      super
```

Comment:
If we write the staged events directly here after `super` then we would always run after any before_commit hooks, and also wouldn't need to add the handler to the transaction records at all so we wouldn't need to worry about that API interface changing (which might be happening? I don't know what is happening [here](https://github.com/rails/rails/blob/d33b1ae495795c84f3f11eb0156c9af35ed34ea3/activerecord/lib/active_record/connection_adapters/abstract/transaction.rb#L199))

```suggestion
      super
      _journaled_transaction_handler.write_events!
```

Reply:
Fair enough! One thing I did have in mind is that not every Rails transaction actually runs callbacks internally, and instead these txn records are handed off to the next transaction in the stack. But you've given me an idea for how we could actually lean on that to pass along the event bundles themselves, and further reduce the number of jobs that get enqueued. I'm going to see if I can try to replicate the behavior I'm talking about and then write some tests around it.